05.wc/wc.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  option, file_names = parse_options
+
+  if file_names.empty?
+    print_counts_for_stdin(option)
+  else
+    count_and_print_files(file_names, option)
+  end
+end
+
+def parse_options
+  opt = OptionParser.new
+  option = {}
+  opt.on('-l') { |l| option[:l] = l }
+  opt.on('-w') { |w| option[:w] = w }
+  opt.on('-c') { |c| option[:c] = c }
+  file_names = opt.parse(ARGV)
+  [option, file_names]
+end
+
+def print_counts_for_stdin(option)
+  file_content = $stdin.read
+  count_content = count(file_content)
+  print_count(count_content, option, '')
+end
+
+def count(file_content)
+  line_count = file_content.lines.count
+  word_count = file_content.split.count
+  byte = file_content.size
+  {
+    line_count:,
+    word_count:,
+    byte:
+  }
+end
+
+def print_count(count_content, option, file_name)
```

Comment:
```suggestion
def print_count(count_content, option)
```

count_content にファイル名を含めるようにすれば引数が減らせそうです。